jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java

```diff
@@ -693,37 +692,26 @@ protected List<Resource> findWebInfLibJars(WebAppContext context)
         throws Exception
     {
         Resource webInf = context.getWebInf();
-        if (webInf == null || !webInf.exists())
+        if (webInf == null)
             return null;
 
-        List<Resource> jarResources = new ArrayList<Resource>();
+        List<Resource> jarResources = new ArrayList<>();
         Resource webInfLib = webInf.resolve("/lib");
-        if (webInfLib.exists() && webInfLib.isDirectory())
+        for (Resource lib: webInfLib) // walk all "lib" hits
         {
-            List<String> files = webInfLib.list();
-            if (files != null)
-            {
-                files.sort(Comparator.naturalOrder());
-            }
-            for (int f = 0; files != null && f < files.size(); f++)
+            if (!lib.exists())
+                continue; // doesn't exist, skip entry
+
+            for (Resource entry: lib.list()) // get contents of "lib" entry
             {
-                try
+                if (FileID.isLibArchive(entry.getFileName()))
                 {
-                    Resource file = webInfLib.resolve(files.get(f));
-                    String fnlc = file.getName().toLowerCase(Locale.ENGLISH);
-                    int dot = fnlc.lastIndexOf('.');
-                    String extension = (dot < 0 ? null : fnlc.substring(dot));
-                    if (extension != null && (extension.equals(".jar") || extension.equals(".zip")))
-                    {
-                        jarResources.add(file);
-                    }
-                }
-                catch (Exception ex)
-                {
-                    LOG.warn("Unable to load WEB-INF file {}", files.get(f), ex);
+                    jarResources.add(entry);
                 }
             }
         }
+
+        jarResources.sort(ResourceCollators.byName(true));
```

Comment:
I think all this block can be done with `Stream` APIs?

Reply:
Yes.  Done.